simple_certmanager/forms.py

```diff
@@ -52,6 +67,46 @@ def save(self, commit=True):
             instance.save()
         return instance
 
+    def clean(self) -> dict[str, Any]:
+        # Errors if the CSR already has a certificate
+        if self.instance.public_certificate:
+            self.add_error(
+                "certificate",
+                "A certificate already exists for this CSR. "
+                "Delete the certificate first.",
+            )
+        # Handle the certificate file upload
+        if "certificate" in self.cleaned_data and self.cleaned_data["certificate"]:
+            uploaded_file = self.cleaned_data["certificate"]
+            try:
+                uploaded_file.seek(0)
+                certificate_data = uploaded_file.read()
+
+                # Load the certificate and throw an error if the certificate is invalid
+                certificate = x509.load_pem_x509_certificate(certificate_data)
```

Comment:
What happens when the certificate is not a PEM file? There is no guarantee that the CA will provide these files as PEM data, often they are some binary format (PKCS?)